examples/transcription/websocket/microphone_stream.rs

```diff
@@ -92,21 +102,55 @@ fn microphone_as_stream() -> FuturesReceiver<Result<Bytes, RecvError>> {
 async fn main() -> Result<(), DeepgramError> {
     let dg = Deepgram::new(env::var("DEEPGRAM_API_KEY").unwrap());
 
-    let mut results = dg
+    let (event_tx, mut event_rx) = tokio::sync::mpsc::channel::<Event>(100);
+
+    // Event handling task
+    tokio::spawn(async move {
+        while let Some(event) = event_rx.recv().await {
+            match event {
+                Event::Open => println!("Connection opened"),
+                Event::Close => println!("Connection closed"),
+                Event::Error(e) => eprintln!("Error occurred: {:?}", e),
+                Event::Result(result) => println!("got: {:?}", result),
+            }
+        }
+    });
+
+    let options = Options::builder()
+        .model(Model::Nova2)
+        .smart_format(true)
+        .language(Language::en_US)
+        .build();
+
+    let (connection, mut response_stream) = dg
         .transcription()
-        .stream_request()
+        .stream_request_with_options(Some(&options))
         .keep_alive()
         .stream(microphone_as_stream())
         .encoding(Encoding::Linear16)
         // TODO Specific to my machine, not general enough example.
         .sample_rate(44100)
         // TODO Specific to my machine, not general enough example.
         .channels(2)
-        .start()
+        .start(event_tx.clone())
         .await?;
 
-    while let Some(result) = results.next().await {
-        println!("got: {:?}", result);
+    let mut count = 0;
+
+    while let Some(response) = response_stream.next().await {
+        // Close the stream after 5 messages are received
+        if count == 5 {
+            // Call finalize after processing the stream
+            connection.finalize(event_tx.clone()).await?;
+
+            // Call  after processing the stream
+            connection.finish(event_tx.clone()).await?;
+        }
+        count += 1;
+        match response {
+            Ok(result) => println!("Transcription result: {:?}", result),
+            Err(e) => eprintln!("Transcription error: {:?}", e),
+        }
```

Comment:
This is a weird mix of high and low level interface.  If the user provides a stream, all the details of managing the stream should Just Work :tm: .  They get back a stream of responses, and when the stream is exhausted (or an error occurs), the stream closes itself before returning None.

In 0.6.1 we can add an interface where they can also be request a LiveTranscriptionHandle (naming open for debate) instead of calling `.stream()`, at which point, they drive the whole thing: sending chunks of audio, receiving responses, finalizing the request, etc.

Reply:
In other words, there should be separate interfaces or getting back a `response_stream` and a `connection`, instead of returning both at the same time.  The proposed interface does not make it clear what the user's responsibilities are vs. what the library handles.